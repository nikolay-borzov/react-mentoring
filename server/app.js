const express = require('express')

const isDevelopment = process.env.NODE_ENV !== 'production'

const app = express()

if (isDevelopment) {
  const webpack = require('webpack')
  const webpackDevMiddleware = require('webpack-dev-middleware')
  const webpackHotMiddleware = require('webpack-hot-middleware')
  const webpackHotServerMiddleware = require('webpack-hot-server-middleware')
  const webpackConfig = require('../webpack')

  const compiler = webpack(webpackConfig)

  app.use(
    webpackDevMiddleware(compiler, {
      historyApiFallback: true
    })
  )
  app.use(
    webpackHotMiddleware(compiler.compilers.find(c => c.name === 'client'))
  )
  app.use(webpackHotServerMiddleware(compiler))
} else {
  const stats = require('../dist/react-loadable.json')
  const serverRenderer = require('../dist/server-renderer').default

  app.use(express.static('dist'))
  app.use(serverRenderer(stats))
}

module.exports = app
