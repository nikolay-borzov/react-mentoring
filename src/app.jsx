import React from 'react'
import { hot } from 'react-hot-loader'
import { ToastContainer, Slide, toast } from 'react-toastify'

import { ErrorBoundary } from './components/error-boundary'
import { SearchContainer } from './pages/search/search-container'
import { FilmContainer } from './pages/film/film-container'

export function App() {
  const searchParams = new URLSearchParams(location.search)

  // TODO: Remove after React Router integration
  const isFilmPage = searchParams.has('id')

  return (
    <ErrorBoundary>
      {isFilmPage ? (
        <FilmContainer filmId={parseInt(searchParams.get('id'), 10)} />
      ) : (
        <SearchContainer />
      )}
      <ToastContainer
        transition={Slide}
        position={toast.POSITION.BOTTOM_CENTER}
        autoClose={false}
      />
    </ErrorBoundary>
  )
}

export default hot(module)(App)
