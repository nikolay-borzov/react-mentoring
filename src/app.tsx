import React from 'react'
import { hot } from 'react-hot-loader'
import { Provider } from 'react-redux'
import { Route, Redirect, Switch } from 'react-router-dom'
import { ToastContainer, Slide, toast } from 'react-toastify'

import { GetLoadable } from './components/helpers'
import { ErrorBoundary } from './components/error-boundary'

// Loadable components
/* istanbul ignore next */
const SearchContainer = GetLoadable(() =>
  import('./pages/search/search-container')
)
/* istanbul ignore next */
const FilmContainer = GetLoadable(() => import('./pages/film/film-container'))
/* istanbul ignore next */
const NotFound = GetLoadable(() => import('./pages/not-found'))

export interface AppContext {
  url?: string
}

interface AppProps {
  Router: Function
  location: string
  context: AppContext
  store: any
}

App.defaultProps = {
  location: null,
  context: null,
  persistor: null
}

export function App({ Router, location, context, store }: AppProps) {
  return (
    <Provider store={store}>
      <ErrorBoundary>
        <Router location={location} context={context}>
          <React.Fragment>
            <Switch>
              <Route path="/search/:search?" component={SearchContainer} />
              <Route path="/film/:id" component={FilmContainer} />
              <Redirect exact from="/" to="/search" />
              <Route component={NotFound} />
            </Switch>
            <ToastContainer
              transition={Slide}
              position={toast.POSITION.BOTTOM_CENTER}
              autoClose={false}
            />
          </React.Fragment>
        </Router>
      </ErrorBoundary>
    </Provider>
  )
}

export default hot(module)(App)
