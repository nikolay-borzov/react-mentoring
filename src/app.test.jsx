import React from 'react'

import {
  itContainsComponent,
  itRendersCorrectlyShallow
} from '../jest/test-helpers'

import { App } from './app'

describe('App component', () => {
  const props = {
    Router: jest.fn(),
    location: '',
    context: {
      url: ''
    },
    store: {
      subscribe: jest.fn(),
      dispatch: jest.fn(),
      getState: jest.fn()
    }
  }

  itRendersCorrectlyShallow(() => <App {...props} />)

  itContainsComponent(() => <App {...props} />, 'ToastContainer')
})
