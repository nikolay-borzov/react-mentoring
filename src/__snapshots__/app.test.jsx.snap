// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [MockFunction],
      "getState": [MockFunction],
    }
  }
>
  <ErrorBoundary>
    <mockConstructor
      context={
        Object {
          "url": "",
        }
      }
      location=""
    >
      <React.Fragment>
        <Switch>
          <Route
            component={[Function]}
            path="/search/:search?"
          />
          <Route
            component={[Function]}
            path="/film/:id"
          />
          <Redirect
            exact={true}
            from="/"
            push={false}
            to="/search"
          />
          <Route
            component={[Function]}
          />
        </Switch>
        <ToastContainer
          autoClose={false}
          bodyClassName={null}
          className={null}
          closeButton={
            <CloseButton
              ariaLabel="close"
            />
          }
          closeOnClick={true}
          draggable={true}
          draggablePercent={80}
          hideProgressBar={false}
          newestOnTop={false}
          pauseOnHover={true}
          pauseOnVisibilityChange={true}
          position="bottom-center"
          progressClassName={null}
          rtl={false}
          style={null}
          toastClassName={null}
          transition={[Function]}
        />
      </React.Fragment>
    </mockConstructor>
  </ErrorBoundary>
</Provider>
`;
